### DISCUSSION TITLE
HTTP3 没有配置证书，不会自签

### DISCUSSION THREAD

**Discussion by xqzr** [XTLS/Xray-core#4087] [General]
它不像 QUIC，不配置证书，会自签证书
```json
{
 "streamSettings": {
  "network": "http",
  "security": "tls",
  "tlsSettings": {
   "alpn": [
    "h3"
   ]
  }
 }
}
 ```
 客户端日志：
 ```log
 [Warning] [1065363489] transport/internet/http: failed to dial to tcp:[fd80:d4a5:e14:206::]:1081 > Put "https://[fd80:d4a5:e14:206::]:1081/": CRYPTO_ERROR 0x170 (remote): tls: unrecognized name
```
客户端收到来自服务端的 `tls: unrecognized name` 错误
https://t.me/projectXray/3994721?thread=3994468

**Comment by RPRX** [comment]
~~好在它与 QUIC 传输层的共同点是也会被删除，最终归宿都是 XHTTP，或者 RAW+TLS H3~~

况且自动自签的话客户端肯定要默认 allowInsecure，这是不合适的

**Reply by xqzr** [reply]
> 自动自签的话客户端肯定要默认 allowInsecure，这是不合适的

我正在**可信**链路中，工作它们

**Reply by RPRX** [reply]
> 我正在**可信**链路中，工作它们

那你自己弄自签加客户端 allowInsecure，为什么要拉上所有人？

**Reply by xqzr** [reply]
> > 我正在**可信**链路中，工作它们
> 
> 那你自己弄自签加客户端 allowInsecure，为什么要拉上所有人？

~~嫌麻烦~~
对于一般公网用户，自动自签的确不合适。也许可以，有个开关（配置项/变量）来控制“自动自签“

**Reply by xqzr** [reply]
> 一般公网用户，自动自签

他们也许不知道，发生了这一切，从而带来风险

**Reply by RPRX** [reply]
> 也许可以，有个开关（配置项/变量）来控制“自动自签“

我觉得这个功能没什么必要，反正客户端 allowInsecure 或者 pin 什么的，你随便签一个证书或者拿个过期证书来就能永久用
